Middleware/utilities/automation_utils.py

```diff
@@ -1,6 +1,17 @@
 import importlib.util
 import os
+import logging
+import traceback
 
+# Base exception for errors occurring within dynamically loaded modules
+class DynamicModuleError(Exception):
```

Comment:
I defined this Exception in this module, so subclasses can inherit from it and throw an exception when PythonModule raised an exception